pkg/lib/scoped/token_retriever.go

```diff
@@ -37,6 +38,10 @@ func (r *BearerTokenRetriever) Retrieve(reference *corev1.ObjectReference) (toke
 
 	if secret == nil {
 		err = fmt.Errorf("the service account does not have any API secret sa=%s/%s", sa.GetNamespace(), sa.GetName())
+		token, _ = requestSAToken(r.kubeclient, sa)
+		if token != "" {
+			err = nil
```

Comment:
could it be worthwhile to add a log line with the underlying error here? I'm thinking it might be good to know from a debugging perspective to understand that there was an attempt to acquire the token but it failed because of ...